aws-lc-rs/src/ed25519.rs

```diff
@@ -116,31 +117,22 @@ impl KeyPair for Ed25519KeyPair {
     }
 }
 
-pub(crate) unsafe fn generate_key(rng: &dyn SecureRandom) -> Result<LcPtr<EVP_PKEY>, ()> {
-    // TODO: Should we drop support for using the provided rng in a minor release?
-
-    let mut seed = [0u8; ED25519_SEED_LEN];
-    rng.fill(&mut seed)?;
-
-    let mut public_key = MaybeUninit::<[u8; ED25519_PUBLIC_KEY_LEN]>::uninit();
-    let mut private_key = MaybeUninit::<[u8; ED25519_PRIVATE_KEY_LEN]>::uninit();
-    ED25519_keypair_from_seed(
-        public_key.as_mut_ptr().cast(),
-        private_key.as_mut_ptr().cast(),
-        seed.as_ptr(),
-    );
-    seed.zeroize();
-
-    // ED25519_keypair_from_seed doesn't set FIPS indicator, and Ed25119 is not approved anyways at this time.
-    // Seems like it could be approved for use in the future per FIPS 186-5 and CMVP guidance.
-    crate::fips::set_fips_service_status_unapproved();
-
-    LcPtr::new(EVP_PKEY_new_raw_private_key(
-        EVP_PKEY_ED25519,
-        null_mut(),
-        private_key.assume_init().as_ptr(),
-        ED25519_PRIVATE_KEY_SEED_LEN,
-    ))
+pub(crate) unsafe fn generate_key() -> Result<LcPtr<EVP_PKEY>, ()> {
+    let pkey_ctx = LcPtr::new(unsafe { EVP_PKEY_CTX_new_id(EVP_PKEY_ED25519, null_mut()) })?;
```

Comment:
I don't have context on the original decision, but majority of the library's code has opted not to define an internal veneer around these API types, to then just be further wrapped by Rust code. I presume this was done to not bake in shared assumptions, or to avoid potential for performance compromises. I'm not entirely sure I have a good answer otherwise.